src/Http/Http/src/Builder/ApplicationBuilder.cs

```diff
@@ -149,4 +167,12 @@ public RequestDelegate Build()
 
         return app;
     }
+
+    private static partial class Log
+    {
+        [LoggerMessage(1, LogLevel.Information,
+            "Request reached the end of the middleware pipeline without being handled by application code {Protocol} {Method} {Scheme}://{Host}{PathBase}{Path} - {StatusCode}",
```

Comment:
@Tratcher What do you think of adding query string to the log? Everything else is here. I think people will find it confusing if the query string is unexplainable missing.

Reply:
Could that be a security issue? Some people put secrets on query strings (e.g. SAS tokens). If someone has a SAS token in their query string, but the base path is incorrect then you could start leaking those active credentials into your logs.

Reply:
There are existing logs that include the query string: https://github.com/dotnet/aspnetcore/blob/e53379914e7ed68eab66f5dac5c8682ac6f3545e/src/Hosting/Hosting/src/Internal/HostingRequestStartingLog.cs#L12

Reply:
Don't. Queries tend to be large, less impactful on routing, and to contain more PII.

Reply:
Protocol isn't relevant to routing either is it?